### PR TITLE
8.0 website menu url translate

### PR DESCRIPTION
**Pull request by rami-wafaie**

### CODE REVIEW

File: website_menu_url_translate/README.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Change Odoo Language to the language of the URL
+* Go to Settings -> Configuration -> Website Settings
+* Click on: Configure website menus
+* but the link of the translated URL in the url Field
```
Comment (simahawk) [inline website_menu_url_translate/README.rst:19]: but/put?

File: website_menu_url_translate/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import openerp
+from openerp.addons.web import http
+from openerp.http import request
+
+
+class Website(openerp.addons.web.controllers.main.Home):
+
+    @http.route('/website/lang/<lang>', type='http',
+                auth="public", website=True, multilang=False)
+    def change_lang(self, lang, r='/', **kwargs):
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:16]: `r` is?
a docstring would be nice to describe what this method gets and does

File: website_menu_url_translate/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        if lang == 'default':
+            lang = request.website.default_lang_code
+            r = '/%s%s' % (lang, r or '/')
+        page_url = ''
+        if lang == request.website.default_lang_code:
+            page_url = str(r)
+        else:
+            r_value = str('/'+str(lang))
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:24]: check PEP8

File: website_menu_url_translate/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    else:
+                        r = ans2[0].src
+                else:
+                    r = ans[0].src
+        redirect = werkzeug.utils.redirect(r or ('/%s' % lang), 303)
+        redirect.set_cookie('website_lang', lang)
+        return redirect
+
+    @http.route('/page/<page:page>', type='http', auth="public", website=True)
+    def page(self, page, **opt):
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:56]: same

File: website_menu_url_translate/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+        # /page/website.XXX --> /page/XXX
+        if page.startswith('website.'):
+            return request.redirect('/page/' + page[8:], code=301)
+        elif '.' not in page:
+            page = 'website.%s' % page
+
+        try:
+            translation_obj = request.env['ir.translation']
+            paths = '/page/' + page.split('.')[1]
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:69]: s/paths/path?

File: website_menu_url_translate/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            # page not found
+            if request.website.is_publisher():
+                page = 'website.page_404'
+            else:
+                return request.registry['ir.http']._handle_exception(e, 404)
+
+        return request.render(page, values)
+
+
+class language_dependent(http.Controller):
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:86]: camelcase

File: website_menu_url_translate/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+    @http.route(['/defaulturl'], type='json', auth="public", website=True)
+    def call(self, menu_id, **post):
```
Comment (simahawk) [inline website_menu_url_translate/controllers/main.py:89]: get_default_url or default_url?

File: website_menu_url_translate/static/src/js/website.contentMenu.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+ *  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+(function () {
+    'use strict';
+
+    var website = openerp.website;
+    var _t = openerp._t;
+
+    website.contentMenu.EditMenuDialog.include({
+		edit_menu: function (ev) {
```
Comment (simahawk) [inline website_menu_url_translate/static/src/js/website.contentMenu.js:12]: fix indent

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_menu_url_translate/__init__.py [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 initOS GmbH. <http://www.initos.com>
```
Comment (nikul-serpentcs) [inline website_menu_url_translate/__init__.py:2]: Remove Copyright from `__init__.py` file

File: website_menu_url_translate/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 initOS GmbH. <http:/www.initos.com>
+# Copyright 2017 GYB IT SOLUTIONS <http:/www.gybitsolutions.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': "Website Menu URL Translation",
+    'category': "Website",
+    'website': "www.gybitsolutions.com",
```
Comment (nikul-serpentcs) [inline website_menu_url_translate/__openerp__.py:9]: @rami-wafaie Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

**Review by nikul-serpentcs — COMMENTED** [review]
Small Change

**Comment by nikul-serpentcs** [comment]
Fix Travis

**Pull request closed without merging**